<launch>
    <!-- Launch file for Livox AVIA LiDAR -->

    <arg name="rviz" default="true"/>

    <rosparam command="load" file="$(find faster_lio)/config/avia_voxel.yaml"/>

    <param name="feature_extract_enable" type="bool" value="0"/>
    <param name="point_filter_num_" type="int" value="3"/>
    <param name="max_iteration" type="int" value="3"/>
    <param name="filter_size_surf" type="double" value="0.5"/>
    <param name="filter_size_map" type="double" value="0.5"/>
    <param name="cube_side_length" type="double" value="1000"/>
    <param name="runtime_pos_log_enable" type="bool" value="1"/>
    <node pkg="faster_lio" type="run_mapping_online" name="laserMapping" output="screen">
        <rosparam file="$(find faster_lio)/config/camera_pinhole_avia.yaml"/>
    </node>

    <group if="$(arg rviz)">
<!--        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz"-->
<!--              args="-d $(find faster_lio)/rviz_cfg/loam_livox.rviz"/>-->
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz"
              args="-d $(find faster_lio)/rviz_cfg/voxel_LIVO.rviz"/>
<!--        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz"-->
<!--              args="-d $(find faster_lio)/rviz_cfg/voxel_mapping.rviz"/>-->
    </group>
    <node pkg="image_transport" type="republish" name="republish"
          args="compressed in:=/left_camera/image raw out:=/left_camera/image" output="screen" respawn="true"/>

    launch-prefix="gdb -ex run --args" launch-prefix="valgrind --leak-check=full --show-leak-kinds=all"

</launch>
